client/blocks/conversation-caterpillar/index.jsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { map, get, last } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/***
+ * Internal dependencies
+ */
+import Gravatar from 'components/gravatar';
+import { getPostCommentsTree } from 'state/comments/selectors';
+
+class ConversationCaterpillarComponent extends React.Component {
+	static propTypes = {
+		blogId: PropTypes.number.isRequired,
+		postId: PropTypes.number.isRequired,
+	};
+
+	render() {
+		const { commentsTree, translate } = this.props;
+		const lastComment = commentsTree[ last( Object.keys( commentsTree ) ) ];
```

Comment:
to expand on my previous comment, assuming commentsTree is a Map, using `last( Object.keys( ... ) )` won't give you the last comment. It'll just give you the last comment based on whatever ordering Object.keys decided to use. That in turn is dictated by how the object was created (keys are returned in creation order). So we probably want a deterministic way to get the "last" comment in a tree.